LaTeX/LaTeX.sublime-syntax

```diff
@@ -4,554 +4,1554 @@
-            4: variable.parameter.function.latex
-            5: punctuation.definition.arguments.latex
+        - meta_scope: meta.environment.math.inline.paren.latex
+        - match: (\\\))
+          scope: string.other.math.latex punctuation.definition.string.end.latex
           pop: true
-        - include: $top_level_main
-    - match: '(?:\s*)((\\)begin)(\{)(\w+[*]?)(\})'
+        - include: math-content
+
+  block-math:
+    - match: \$\$
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: meta.environment.math.block.dollar.latex
+        - match: \$\$
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+        - include: math-content
+
+    - match: '(\\\[)'
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: meta.environment.math.block.bracket.latex
+        - match: '(\\\])'
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+        - include: math-content
+
+    - match: '((\\)begin)(\{)\s*((?:align|alignat|aligned|alignedat|displaymath|eqnarray|equation|flalign|gather|gathered|math|multline|x?xalignat|split)\*?)\s*(\})'
       captures:
         1: support.function.be.latex
-        2: punctuation.definition.function.latex
-        3: punctuation.definition.arguments.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
         4: variable.parameter.function.latex
-        5: punctuation.definition.arguments.latex
+        5: punctuation.definition.group.brace.end.latex
       push:
-        - meta_scope: meta.function.environment.general.latex
-        - match: '((\\)end)(\{)(\4)(\})(?:\s*\n)?'
+        - meta_scope: meta.environment.math.block.be.latex
+        - match: '((\\)end)(\{)\s*(\4)\s*(\})'
           captures:
             1: support.function.be.latex
-            2: punctuation.definition.function.latex
-            3: punctuation.definition.arguments.latex
+            2: punctuation.definition.backslash.latex
+            3: punctuation.definition.group.brace.begin.latex
             4: variable.parameter.function.latex
-            5: punctuation.definition.arguments.latex
+            5: punctuation.definition.group.brace.end.latex
           pop: true
-        - include: $top_level_main
-    - match: (\\)(newcommand|renewcommand)\b
-      scope: storage.type.function.latex
+        - include: math-content
+
+  url:
+    - match: '(?:\s*)((\\)(?:url|href))(\{)([^}]*)(\})'
+      scope: meta.function.link.url.latex
+      captures:
+        1: support.function.url.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+        4: markup.underline.link.latex
+        5: punctuation.definition.group.brace.end.latex
+
+  invalid-quotations:
```

Comment:
This also highlights valid quotations like ``"`string"'`` to invalid. This supported by the [german babel package](https://en.wikibooks.org/wiki/LaTeX/Internationalization#German)